students/timm67/session02/print_grid.py

```diff
@@ -1,10 +1,82 @@
+#!/usr/bin/env python3
+
 """
 Tim Meese
 Au2018-Py210B
 Print Grid Exercise
 """
 
-#
-# Part 1: print a fixed size grid (2x2)
-#
+_star = '*'
+_line = '-'
+_intersect = '+'
+_vertline = '|'
+_space = ' '
+
+def print_grid(n):
+    global _star, _line, _intersect, _vertline, _space
+    outlines = []
+    cur_row = 0
+    cur_col = 0
+
+    if n < 11:
+        n = n * 2 + 1
+
+    for j in range(n):
+        grid = ''
+        cur_col = 0
+        for i in range(n):
+            # when to output intersection 
+            if (cur_row % (n // 2)) == 0 and (cur_col % (n // 2)) == 0:
+                grid += _intersect
+            elif (cur_row % (n // 2)) == 0:
+                grid += _line
+            elif (cur_col % (n // 2) == 0):
+                grid += _vertline
+            else:
+                grid += _space
+            cur_col += 1
+        cur_row += 1
+        outlines.append(grid)
+
+    for line in outlines:
+        print(line)
+
+def print_grid2(num_columns, column_width):
+    global _star, _line, _intersect, _vertline, _space
+    outlines = []
+    cur_row = 0
+    cur_col = 0
+
+    for j in range(num_columns*column_width+1):
+        grid = ''
+        cur_col = 0
+        for i in range(num_columns*column_width+1):
+            # when to output intersection 
+            if (cur_row % column_width) == 0 and (cur_col % column_width) == 0:
+                grid += _intersect
+            elif (cur_row % column_width) == 0:
+                grid += _line
+            elif (cur_col % column_width) == 0:
+                grid += _vertline
+            else:
+                grid += _space
+            cur_col += 1
+        cur_row += 1
```

Comment:
Why use `cur_row` and `cur_col`, when `i` and `j` seem to be the same corresponding values?

Reply:
Yes, true. Could have used cur_row, and cur_col as iterators. I think I have implemented what was asked for in the assignment and run all the examples in the 'how to' document.